client/gutenberg/editor/index.js

```diff
@@ -24,52 +22,78 @@ export default function () {
 		clientRender
 	);
 
-	page( '/block-editor/post', siteSelection, sites, makeLayout, clientRender );
+	page( '/post', siteSelection, sites, makeLayout, clientRender );
+	page( '/post/new', () => page.redirect( '/post' ) ); // redirect from beep-beep-boop
 	page(
-		'/block-editor/post/:site/:post?',
+		'/post/:site/:post?',
 		siteSelection,
 		redirect,
 		authenticate,
 		post,
 		makeLayout,
 		clientRender
 	);
-	page( '/block-editor/post/:site?', siteSelection, redirect, makeLayout, clientRender );
+	page.exit( '/post/:site?/:post?', exitPost );
+	page( '/post/:site?', siteSelection, redirect, makeLayout, clientRender );
 
-	page( '/block-editor/page', siteSelection, sites, makeLayout, clientRender );
+	page( '/page', siteSelection, sites, makeLayout, clientRender );
+	page( '/page/new', () => page.redirect( '/post' ) ); // redirect from beep-beep-boop
 	page(
-		'/block-editor/page/:site/:post?',
+		'/page/:site/:post?',
 		siteSelection,
 		redirect,
 		authenticate,
 		post,
 		makeLayout,
 		clientRender
 	);
-	page( '/block-editor/page/:site?', siteSelection, redirect, makeLayout, clientRender );
+	page.exit( '/page/:site?/:post?', exitPost );
+	page( '/page/:site?', siteSelection, redirect, makeLayout, clientRender );
 
 	if ( config.isEnabled( 'manage/custom-post-types' ) ) {
-		page( '/block-editor/edit/:customPostType', siteSelection, sites, makeLayout, clientRender );
+		page( '/edit/:customPostType', siteSelection, sites, makeLayout, clientRender );
 		page(
-			'/block-editor/edit/:customPostType/:site/:post?',
+			'/edit/:customPostType/:site/:post?',
 			siteSelection,
 			redirect,
 			authenticate,
 			post,
 			makeLayout,
 			clientRender
 		);
-		page(
-			'/block-editor/edit/:customPostType/:site?',
-			siteSelection,
-			redirect,
-			makeLayout,
-			clientRender
-		);
+		page( '/edit/:customPostType/:site?', siteSelection, redirect, makeLayout, clientRender );
 	}
 
-	page( '/block-editor/*/*', '/block-editor/post' );
-	page( '/block-editor/:site', ( context ) =>
-		page.redirect( `/block-editor/post/${ context.params.site }` )
-	);
+	/*
+	 * Redirecto the old `/block-editor` routes to the default routes.
+	 */
+	page( '/block-editor/', '/post' );
+	page( '/block-editor/post/', '/post' );
+	page( '/block-editor/post/:site/:post?', ( { params = {} } ) => {
+		const { site, post: postId } = params;
+		if ( postId ) {
+			return page.redirect( `/post/${ site }/${ postId }` );
+		}
+		page.redirect( `/post/${ site }/` );
+	} );
+
+	page( '/block-editor/page/', '/page' );
+	page( '/block-editor/page/:site/:page?', ( { params = {} } ) => {
+		const { site, page: pageId } = params;
+		if ( pageId ) {
+			return page.redirect( `/page/${ site }/${ pageId }` );
+		}
+		page.redirect( `/page/${ site }/` );
+	} );
+
+	if ( config.isEnabled( 'manage/custom-post-types' ) ) {
+		page( '/block-editor/edit/:customPostType/:site/:post?', ( { params = {} } ) => {
+			const { customPostType, site, page: postId } = params;
```

Comment:
I thinks this is causing the odd behavior noted by @allilevine in https://github.com/Automattic/wp-calypso/pull/46006#discussion_r503567121

```suggestion
			const { customPostType, site, post: postId } = params;
```

Reply:
Thanks for this, Miguel.